### PR TITLE
Updates configuring.md to be more clear on framework defaults

### PR DESCRIPTION
**Pull request by systemnate**
### Summary

I was confused in looking through the documentation.  It seems that for brand new applications, `config.action_controller.default_protect_from_forgery` is `true`.  Only when upgrading, and before you flip the new framework defaults, it is `false`.

### Other Information

Currently, the docs say:
> config.action_controller.default_protect_from_forgery determines whether forgery protection is added on ActionController::Base. This is false by default.

However, a little bit further it says (in https://edgeguides.rubyonrails.org/configuring.html#with-5-2):

> config.action_controller.default_protect_from_forgery: true

Additionally, `config/initializers/new_framework_defaults_5_2.rb` says:

> This file contains migration options to ease your Rails 5.2 upgrade.
> Once upgraded flip defaults one by one to migrate to the new default.
...
> Rails.application.config.action_controller.default_protect_from_forgery = true

It seems to me that the default is actually `true`.

### CODE REVIEW

**Comment by bogdanvlviv** [comment]
By default `default_protect_from_forgery` is set to `false`, see https://github.com/rails/rails/blob/b53b188c5ea4db8ca3c1c80c21a9bc1d653a4f08/actionpack/lib/action_controller/metal/request_forgery_protection.rb#L99

To apply the framework defaults we use  [`load_defaults`](https://github.com/rails/rails/blob/966d3a7bf2c08c5c20213d29686cbb409a07734a/railties/lib/rails/application/configuration.rb#L75) method.
[The guide explaines](https://edgeguides.rubyonrails.org/configuring.html#results-of-load-defaults) what defaults are applied when you call the method with ` '5.2'` or `'6.0'`, etc. (Note that every version includes all defaults of previous versions)

**Comment by systemnate** [comment]
Thanks - I did not see that.  I'm still not completely clear.  What calls `#load_defaults`?  If you create a brand new Rails 6 application using `rails new`, what is the value of `default_protect_from_forgery`?  Are you saying it is `false`?

Thanks in advance!  Somehow I am not understanding what I read from the docs.

**Comment by bogdanvlviv** [comment]
@systemnate Please read https://github.com/rails/rails/pull/34855#issuecomment-451287368. I also opened PR to clarify docs of `load_defaults` in https://github.com/rails/rails/pull/37595. I hope it would help users.

**Comment by systemnate** [comment]
Thanks for the additional history and new PR.  I think it is an improvement; however, I would still be confused if I were to read it for the first time again.  In the comment you linked to, @rafaelfranca  said:

> Sorry but I disagree. We should talk about the default value of the configuration when no railtie code is applied. So the current documentation is correct. If we tell that the default is true for config.action_controller.default_protect_from_forgery and the user see that config being set to true in the application they will think it is safe to remove when it is not.

Can you help me understand why it is essential to talk about the defaults when no railtie code is applied?  It appears that the reason is to avoid someone removing `protect_from_forgery` when they should not.  For example, a user on a version of Rails before 5.2 might think they could remove it when they should not.  To me, this confusion can by saying something like "The default for Rails versions >= 5.2 is true when `config.load_defaults "5.2"` (or higher) is used; otherwise, the value is false".  I believe this would eliminate the ambiguity, especially for someone reading this guide for the first time.

Existing verbiage:
> config.action_controller.default_protect_from_forgery determines whether forgery protection is added on ActionController:Base. This is false by default.

File: guides/source/configuring.md [hunk 1]
```diff
@@ -440,7 +440,7 @@ The schema dumper adds two additional configuration options:
 
 * `config.action_controller.per_form_csrf_tokens` configures whether CSRF tokens are only valid for the method/action they were generated for.
 
-* `config.action_controller.default_protect_from_forgery` determines whether forgery protection is added on `ActionController::Base`. This is false by default.
+* `config.action_controller.default_protect_from_forgery` determines whether forgery protection is added on `ActionController::Base`. For new applications, the default is true.
```
Comment (vipulnsward) [inline guides/source/configuring.md]: We can follow other places and simply say:

> This is true by default.

We change a lot of these options, and in this guide we should just mention what the defaults are.

**Pull request closed without merging**